src/pika_slot.cc

```diff
@@ -911,3 +964,455 @@ void SlotsScanCmd::Do() {
+        res_.SetRes(CmdRes::kInvalidInt);
+        return;
+    }
+
+    std::string str_keys_num = *it++;
+    if (!slash::string2l(str_keys_num.data(), str_keys_num.size(), &keys_num_) || keys_num_ < 0) {
+        res_.SetRes(CmdRes::kInvalidInt);
+        return;
+    }
+    return;
+}
+
+void SlotsMgrtTagSlotAsyncCmd::Do() {
+    bool ret = g_pika_server->SlotsMigrateBatch(dest_ip_, dest_port_, timeout_ms_, slot_num_, keys_num_);
+    if (!ret) {
+        LOG(WARNING) << "Slot batch migrate keys error";
+        res_.SetRes(CmdRes::kErrOther, "Slot batch migrating keys error");
+        return;
+    }
+
+    int64_t moved = 0, remained = 0;
+    ret = g_pika_server->GetSlotsMigrateResul(&moved, &remained);
+    if (ret){
+        res_.AppendArrayLen(2);
+        res_.AppendInteger(moved);
+        res_.AppendInteger(remained);
+    } else {
+        LOG(WARNING) << "Slot batch migrate keys get result error";
+        res_.SetRes(CmdRes::kErrOther, "Slot batch migrating keys get result error");
+        return;
+    }
+    return;
+}
+
+void SlotsMgrtExecWrapperCmd::DoInitial(PikaCmdArgsType &argv, const CmdInfo* const ptr_info) {
+    if (!ptr_info->CheckArg(argv.size())) {
+        res_.SetRes(CmdRes::kWrongNum, kCmdNameSlotsMgrtExecWrapper);
+    }
+    PikaCmdArgsType::iterator it = argv.begin() + 1;
+    key_ = slash::StringToLower(*it++);
+    return;
+}
+
+void SlotsMgrtExecWrapperCmd::Do() {
+    res_.AppendArrayLen(2);
+    int ret = g_pika_server->SlotsMigrateOne(key_);
+    switch (ret) {
+    case 0:
+        res_.AppendInteger(0);
+        res_.AppendInteger(0);
+        return;
+    case 1:
+        res_.AppendInteger(1);
+        res_.AppendInteger(1);
+        return;
+    default:
+        res_.AppendInteger(-1);
+        res_.AppendInteger(-1);
+        return;
+    }
+    return;
+}
+
+void SlotsMgrtAsyncStatusCmd::DoInitial(PikaCmdArgsType &argv, const CmdInfo* const ptr_info) {
+    if (!ptr_info->CheckArg(argv.size())) {
+        res_.SetRes(CmdRes::kWrongNum, kCmdNameSlotsMgrtExecWrapper);
+    }
+    return;
+}
+
+void SlotsMgrtAsyncStatusCmd::Do() {
+    std::string status;
+    std::string ip;
+    int64_t port, slot, moved, remained;
+    bool migrating;
+    g_pika_server->GetSlotsMgrtSenderStatus(&ip, &port, &slot, &migrating, &moved, &remained);
+    std::string mstatus = migrating ? "yes" : "no";
+    std::stringstream tmp_stream;
+    tmp_stream << "dest server: " << ip << ":" << std::to_string(port) << "\r\n";
+    tmp_stream << "slot number: " << std::to_string(slot) << "\r\n";
+    tmp_stream << "migrating  : " << mstatus << "\r\n";
+    tmp_stream << "moved keys : " << std::to_string(moved) << "\r\n";
+    tmp_stream << "remain keys: " << std::to_string(remained) << "\r\n";
+    status.append(tmp_stream.str());
+    res_.AppendStringLen(status.size());
+    res_.AppendContent(status);
+    return;
+}
```

Comment:
errrr, this command's output is like:
"dest server: 10.11.11.101:9222\r\nslot number: 39\r\nmigrating  : no\r\nmoved keys : 128\r\nremain keys: 0\r\n"

could it be more human readable?

Reply:
@left2right maybe you need to use bulk strings in this case